tests/test_consdbClient.py

```diff
@@ -0,0 +1,140 @@
+# This file is part of summit_utils.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+import responses
+
+from lsst.summit.utils import ConsDbClient
+
+
+@pytest.fixture
+def client():
+    return ConsDbClient("http://example.com/consdb")
+
+
+def test_table_name(client):
+    instrument = "latiss"
+    obs_type = "exposure"
+    assert client.compute_flexible_metadata_table_name(instrument, obs_type) == "cdb_latiss.exposure_flexdata"
+
+
+@responses.activate
+def test_add_flexible_metadata_key(client):
+    instrument = "latiss"
+    obs_type = "exposure"
+    responses.post(
+        "http://example.com/consdb/flex/latiss/exposure/addkey",
+        json={
+            "message": "Key added to flexible metadata",
+            "key": "foo",
+            "instrument": "latiss",
+            "obs_type": "exposure",
+        },
+        match=[
+            responses.matchers.json_params_matcher({"key": "foo", "dtype": "bool", "doc": "bool key"}),
+        ],
+    )
+    responses.post(
+        "http://example.com/consdb/flex/latiss/exposure/addkey",
+        json={
+            "message": "Key added to flexible metadata",
+            "key": "bar",
+            "instrument": "latiss",
+            "obs_type": "exposure",
+        },
+        match=[
+            responses.matchers.json_params_matcher({"key": "bar", "dtype": "int", "doc": "int key"}),
+        ],
+    )
+    responses.post(
+        "http://example.com/consdb/flex/bad_instrument/exposure/addkey",
+        status=404,
+        json={"message": "Unknown instrument", "value": "bad_instrument", "valid": ["latiss"]},
+    )
+    responses.post(
+        "http://example.com/consdb/flex/latiss/bas_obs_type/addkey",
```

Comment:
Probably doesn't matter, but did you mean to write `bas_obs_type` or `bad_obs_type`?